globus_action_provider_tools/validation.py

```diff
@@ -43,14 +43,42 @@ def request_validator(request: ValidationRequest) -> ValidationResult:
 
 
 def validate_data(
+    data: Dict[str, Any],
+    validator: jsonschema.protocols.Validator,
+    validation_error_obscuring: bool = True,
+) -> ValidationResult:
+    if validation_error_obscuring:
+        return _validate_data_omitting_request_data(data, validator)
+    else:
+        return _validate_data_allowing_request_data(data, validator)
+
+
+def _validate_data_allowing_request_data(
+    data: Dict[str, Any], validator: jsonschema.protocols.Validator
+) -> ValidationResult:
+
+    error_messages = []
+    for error in validator.iter_errors(data):
+        if error.path:
+            # Elements of the error path may be integers or other non-string types,
+            # but we need strings for use with join()
+            error_path_for_message = ".".join([str(x) for x in error.path])
+            error_message = f"'{error_path_for_message}' invalid due to {error.message}"
```

Comment:
Might want to share/mirror the other path formatting construction? https://github.com/globus/action-provider-tools/blob/9021c854a7604635ec9e7c40df06d7feebbb1477/globus_action_provider_tools/flask/helpers.py#L224-L230

Reply:
Hmm, or maybe we can get this automatically from python-jsonschema? https://github.com/python-jsonschema/jsonschema/blob/9df3ab01c275d1a6cf46a95c7b7e90942cbef162/jsonschema/exceptions.py#L141-L149

Reply:
Applied this suggestion; pydantic & jsonschema errors now run through a standard `format_validation_error` function call which enforces standardization of fields, location, and default messaging.